uvtools/dspec.py

```diff
@@ -450,12 +450,17 @@ def gen_window(window, N, alpha=0.5, edgecut_low=0, edgecut_hi=0, **kwargs):
         edgecut_hi = None
     if window in ['none', None, 'None', 'boxcar', 'tophat']:
         w[edgecut_low:edgecut_hi] = windows.boxcar(N - Ncut)
-    elif window in ['blackmanharris', 'blackman-harris']:
+    elif window in ['blackmanharris', 'blackman-harris', 'bh', 'bh4']:
         w[edgecut_low:edgecut_hi] =  windows.blackmanharris(N - Ncut)
     elif window in ['hanning', 'hann']:
         w[edgecut_low:edgecut_hi] =  windows.hann(N - Ncut)
     elif window == 'tukey':
         w[edgecut_low:edgecut_hi] =  windows.tukey(N - Ncut, alpha)
+    elif window in ['blackmanharris-7term', 'blackman-harris-7term', 'bh7']:
+        # https://ieeexplore.ieee.org/document/940309
+        a_k = [0.27105140069342, -0.43329793923448, 0.21812299954311, -0.06592544638803,
+               0.01081174209837, -0.00077658482522, 0.00001388721735]
+        w[edgecut_low:edgecut_hi] = np.sum([[ak * np.cos(2 * np.pi * k * n / float(N - Ncut - 1)) for n in range(N - Ncut)] for k, ak in enumerate(a_k)], axis=0)
```

Comment:
This line will produce division by zero if `N` is 1 (or if `N - Ncut` is 1). We should check for this case, and return an array with a value of `1.0` (the same behavior as for the scipy Blackman-Harris window).

Reply:
we do this a few lines above

Reply:
If `N` is 1, and `Ncut` is 0 (as it would be if `edgecut_low` and `edgecut_hi` are not specified), then the ValueError is not raised, and so we get a ZeroDivisionError:
```
In [2]: gen_window("bh7", 1)
---------------------------------------------------------------------------
ZeroDivisionError                         Traceback (most recent call last)
<ipython-input-2-c6dafc316842> in <module>
----> 1 gen_window("bh7", 1)

~/Documents/school/penn/software/uvtools/uvtools/dspec.py in gen_window(window, N, alpha, edgecut_low, edgecut_hi, **kwargs)
    461         a_k = [0.27105140069342, -0.43329793923448, 0.21812299954311, -0.06592544638803,
    462                0.01081174209837, -0.00077658482522, 0.00001388721735]
--> 463         w[edgecut_low:edgecut_hi] = np.sum([[ak * np.cos(2 * np.pi * k * n / float(N - Ncut - 1)) for n in range(N - Ncut)] for k, ak in enumerate(a_k)], axis=0)
    464     else:
    465         try:

~/Documents/school/penn/software/uvtools/uvtools/dspec.py in <listcomp>(.0)
    461         a_k = [0.27105140069342, -0.43329793923448, 0.21812299954311, -0.06592544638803,
    462                0.01081174209837, -0.00077658482522, 0.00001388721735]
--> 463         w[edgecut_low:edgecut_hi] = np.sum([[ak * np.cos(2 * np.pi * k * n / float(N - Ncut - 1)) for n in range(N - Ncut)] for k, ak in enumerate(a_k)], axis=0)
    464     else:
    465         try:

~/Documents/school/penn/software/uvtools/uvtools/dspec.py in <listcomp>(.0)
    461         a_k = [0.27105140069342, -0.43329793923448, 0.21812299954311, -0.06592544638803,
    462                0.01081174209837, -0.00077658482522, 0.00001388721735]
--> 463         w[edgecut_low:edgecut_hi] = np.sum([[ak * np.cos(2 * np.pi * k * n / float(N - Ncut - 1)) for n in range(N - Ncut)] for k, ak in enumerate(a_k)], axis=0)
    464     else:
    465         try:

ZeroDivisionError: float division by zero
```

I think that requesting a window of size 1 is not a crazy thing, so I'd like to return an array with a single value of 1. The scipy functions behave this way (and will even return an empty array for N=0, though I don't necessarily think we need to support that), so I'd like to keep that parallelism so the user has more consistency.

Reply:
Sounds good.